.github/workflows/e2e.maven.workflow_dispatch.main.default.slsa3.yml

```diff
@@ -0,0 +1,98 @@
+name: Maven e2e test - simple # TODO: Remove name
+on:
+  - workflow_dispatch
+# TODO: Un-comment this     
+#  schedule:
+#    - cron: "0 6 * * *"
+#  workflow_dispatch:
+#
+permissions: read-all
+
+env:
+  GH_TOKEN: ${{ github.token }}
+
+jobs:
+  bootstrap:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
+    permissions:
+      contents: write
+    outputs:
+      artifact-version: ${{ steps.maven-push.outputs.artifact-version }}
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - env:
+          PACKAGE_DIR: ./e2e/maven
+        id: maven-push
+        run: ./.github/workflows/scripts/e2e-maven-push.sh
+
+  if-bootstrap-failed:
+    runs-on: ubuntu-latest
+    needs: [bootstrap]
+    if: always() && (github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && !inputs.trigger_build)) && needs.bootstrap.result != 'success'
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - run: ./.github/workflows/scripts/e2e-report-failure.sh
+  
+  # Main workflow
+  ################################################################################
+  # Shim determines if the rest of the workflow should run.
+  # NOTE: it should only use the `if` to determine this and all downstream jobs
+  #      should depend on this job.
+  shim:
+    # NOTE: this must be kept in sync with the if-failed job.
+    if: github.event_name == 'workflow_dispatch' && inputs.trigger_build
+    runs-on: ubuntu-latest
+    steps:
+      - run: |
+          echo "event: ${GITHUB_EVENT_NAME}"
+          echo "ref: ${GITHUB_REF}"
+
+  build:
+    permissions:
+      id-token: write # For signing.
+      contents: read # For repo checkout of private repos.
+      actions: read # For getting workflow run on private repos.
+    uses: AdamKorcz/slsa-github-generator/.github/workflows/builder_maven_slsa3.yml@maven-builder-test-updates
+    with:
+      directory: ./e2e/maven
+
+  verify:
+    runs-on: ubuntu-latest
+    needs: [build, bootstrap]
```

Comment:
only needs to depend on `[build]`, since build depends on shim

Reply:
Right. Actually `boostrap` will run in a separate workflow run so it's important that you *don't* depend on it here.

Reply:
I believe we won't be able to get the version output if we don't `need` it.

Reply:
on a push, you'll get the version from the github event?
on workflow_dispatch, I'll let @ianlewis comment he's more familiar with the bootstrap code

Reply:
I specifically output the artifact version in bootstrap:

```yaml
jobs:
  bootstrap:
    runs-on: ubuntu-latest
    if: github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
    permissions:
      contents: write
    outputs:
      artifact-version: ${{ steps.maven-push.outputs.artifact-version }} <---------------- Here
    steps:
      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
      - env:
          PACKAGE_DIR: ./e2e/maven/workflow_dispatch
        id: maven-push
        run: ./.github/workflows/scripts/e2e-maven-push.sh
```

This may not be the convention of the tests. LMK.

Reply:
@ianlewis please comment: We don't need the output version for push / tag events. But how do you handle that for workflow_dispatch?

Reply:
I think you don't want to output the version from the bootstrap. Bootstrap is designed to run in a seprarate workflow run.

- `schedule` or `workflow_dispatch` triggers workflow run A at commit X (some random commit)
  - Run `bootstrap`
    - Update the maven package version and commit
    - Create a new `workflow_dispatch` at the new commit (this is critical)
  - Don't run e2e tests steps (blocked by `shim`)
- `workflow_dispatch` from workflow A triggers workflow B at commit Y (our commit with package version updated)
  - Don't run `bootstrap`
  - Run the e2e test steps

In order to get the version could you run a command to get it from the package itself once it's been checked out?  This is how we do it for the Node.js tests. Maybe either running `mvn` to output it or `grep` (or some other xml parsing tool) to get it from the `pom.xml`?
https://github.com/slsa-framework/example-package/blob/bee082ddeb9ab63ffebd5808383308424fb01175/.github/workflows/scripts/e2e.nodejs.default.verify.sh#L66-L68

Reply:
Should be fixed in https://github.com/slsa-framework/example-package/pull/253/commits/69ff36995853f4adb8c6eaffa9bce63336453d1a. The verify script now gets the version from the `pom.xml` instead of from an input to the workflow job.